### DISCUSSION TITLE
Minimum Setpoint Deadband in the Thermostat dialog and card

### DISCUSSION THREAD

**Discussion by dbolger** [home-assistant/frontend#18456] [Dashboard]
Hey there - I've been working on implementing the minimum setpoint deadband value into the Matter integration recently. While messing with the UI to test it, I think it would be interesting to integrate this value to show (or prevent) the user from ever entering a value within the deadband.

For example (pardon my mockup), when a user selects either the heat or the cool target temperature, a dynamic visual between the two temperature ranges will appear to show the user where the deadband lives.

On my thermostat, my minimum setpoint deadband is 3°F, meaning if I try to set the target cool temperature on the right to 73°F or 74°F, I will get an error. It's basically a 'dead zone', meaning these two target temperatures have to have 3°F of range between them.

![image](https://github.com/home-assistant/frontend/assets/37427166/32ac1bbe-c855-4ea6-b0d5-cac3921f0013)

**Comment by dbolger** [comment]
After some more thinking, the deadband would probably be the inverse of the image, where there's a 3°F buffer (or whatever the thermostat's minimum is) on each side of the temperature slider.

When the heat temperature is selected:
![image](https://github.com/home-assistant/frontend/assets/37427166/71d4ae33-32fd-45d9-b203-8d8c513738be)

Cool temperature:
![image](https://github.com/home-assistant/frontend/assets/37427166/038ae04c-a37e-4124-b7fb-0afdf48f169e)

**Comment by karwosts** [comment]
I guess before the frontend could do this, there would need to be an attribute available in the climate entity that describes the properties of the deadband? I don't believe there is such a thing that the frontend could read. Have you considered raising this in [home-assistant/architecture?](https://github.com/home-assistant/architecture/discussions) ?

**Reply by dbolger** [reply]
I haven't raised it in the architecture side, but I have been playing with the implementation locally. I imagine it would be an optional attribute.

**Reply by dbolger** [reply]
I updated the description on my follow up mock-ups.
